enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/StoreResource.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.catchup.storecopy;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.channels.ReadableByteChannel;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.io.pagecache.PagedFile;
+
+class StoreResource implements Closeable
+{
+    private final File file;
+    private final String path;
+    private final int recordSize;
+    private final PageCache pageCache;
+    private final FileSystemAbstraction fs;
+
+    private ReadableByteChannel channel;
+
+    StoreResource( File file, String relativePath, int recordSize, PageCache pageCache, FileSystemAbstraction fs )
+    {
+        this.file = file;
+        this.path = relativePath;
+        this.recordSize = recordSize;
+        this.pageCache = pageCache;
+        this.fs = fs;
+    }
+
+    ReadableByteChannel open() throws IOException
+    {
+        Optional<PagedFile> existingMapping = pageCache.getExistingMapping( file );
+        if ( existingMapping.isPresent() )
```

Comment:
I think this can be changed to
`return existingMapping.map(pFile -> pFile.openReadableBytChannel()).orElse(fs.open(file, "r"));`

Reply:
There is a trw in there as well, which does the close().